dotCMS/src/main/java/com/dotmarketing/portlets/rules/conditionlet/Conditionlet.java

```diff
@@ -3,12 +3,50 @@
 import com.dotmarketing.portlets.rules.RuleComponentDefinition;
 import com.dotmarketing.portlets.rules.RuleComponentInstance;
 import com.dotmarketing.portlets.rules.parameter.ParameterDefinition;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import java.util.Map;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
 
 public abstract class Conditionlet<T extends RuleComponentInstance> extends RuleComponentDefinition<T> {
 
     public static final String COMPARISON_KEY = "comparison";
 
+    /**
+     * @deprecated Do not use this method directly, it exists only for the <em>Jackson</em>-binding
+     * infrastructure
+     */
+    protected Conditionlet(String id, String i18nKey, ParameterDefinition... parameterDefinitions) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameter 'parameterDefinitions' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4925354)